### PR TITLE
EG-1373 Explicit break marker for bullet point lists

### PR DESCRIPTION
**Pull request by woollett-r**

### CODE REVIEW

File: app/services/BulletPointBuilder.scala [hunk 1]
```diff
@@ -26,7 +26,9 @@ import scala.annotation.tailrec
 object BulletPointBuilder {
 
   val notSpaceRegex: Regex = """([^ ]+)""".r
-  val matchLimit: Int = 3
+  val matchLimitEnglish: Int = 3
+  val matchLimitWelsh: Int = 1
+  val explicitBreak: String = "[break]"
```
Comment (cmrhmrc) [inline app/services/BulletPointBuilder.scala:31]: Convention: Capitalised initial char non constants

File: app/services/BulletPointBuilder.scala [hunk 2]
```diff
@@ -330,11 +332,52 @@ object BulletPointBuilder {
   }
 
   def matchPhrases(p1: Phrase, p2: Phrase): Boolean = {
```
Comment (cmrhmrc) [inline app/services/BulletPointBuilder.scala:334]: uneeded braces

File: app/services/BulletPointBuilder.scala [hunk 3]
```diff
@@ -330,11 +332,52 @@ object BulletPointBuilder {
...
-    // Apply matching logic to English text as this is how Ocelot works currently
-    val (p1NoOfWordsToDisplay, p2NoOfWordsToDisplay, matchedWords) = matchInstructionText(p1.english, p2.english)
 
-    // Matching instructions must have matching leading text followed dissimilar trailing text
-    matchedWords.size >= matchLimit && (matchedWords.size < p1NoOfWordsToDisplay) && (matchedWords.size < p2NoOfWordsToDisplay)
+    if(useExplicitMatch(p1, p2)) {
```
Comment (cmrhmrc) [inline app/services/BulletPointBuilder.scala:336]: Both implicitMatchPhrases() and explicitMatchPhrases() could be removed and their single line content be substituted for their sole usage in matchPhrases(). Would reduce code and make the codes intention clearer

File: app/services/BulletPointBuilder.scala [hunk 4]
```diff
@@ -330,11 +332,52 @@ object BulletPointBuilder {
...
+      explicitMatchPhrases(p1, p2)
+    } else {
+      implicitMatchPhrases(p1, p2)
+    }
+
+  }
+
+  def useExplicitMatch(p1: Phrase, p2: Phrase): Boolean = {
```
Comment (cmrhmrc) [inline app/services/BulletPointBuilder.scala:344]: Extra braces not required

File: app/services/UIBuilder.scala [hunk 5]
```diff
@@ -256,22 +258,47 @@ class UIBuilder {
   private def createBulletPointListComponents(phraseGroup: Seq[Phrase])
```
Comment (cmrhmrc) [inline app/services/UIBuilder.scala:258]: Braces not needed

File: app/services/BulletPointBuilder.scala (hunk 1 above)
Comment (woollett-r) [reply app/services/BulletPointBuilder.scala:31]: Done

File: app/services/BulletPointBuilder.scala (hunk 3 above)
Comment (woollett-r) [reply app/services/BulletPointBuilder.scala:336]: Done.

File: app/services/BulletPointBuilder.scala (hunk 4 above)
Comment (woollett-r) [reply app/services/BulletPointBuilder.scala:344]: Done

File: app/services/UIBuilder.scala (hunk 5 above)
Comment (woollett-r) [reply app/services/UIBuilder.scala:258]: Done

File: app/services/BulletPointBuilder.scala (hunk 2 above)
Comment (woollett-r) [reply app/services/BulletPointBuilder.scala:334]: I've put the methods explicitMatch and implicitMatch inside matchPhrases so the brackets are now necessary.

**Pull request merged**